### ISSUE TITLE
Creating slotFills for extensions to hook into

### ISSUE THREAD

**Issue by senadir** [woocommerce/woocommerce-gutenberg-products-block#2125]
This is a later issue, I'm opening so I don't forget and so that we can start a discussion when times come.

We should start considering what kind of slot we want to include in our code so that third party can integrate with it.
for reference, a slotFill pattern is something used in Gutenberg for extending things (like adding settings and menu options)
https://github.com/WordPress/gutenberg/tree/master/packages/components/src/slot-fill/

With that, we should think about extending it, some ideas:
- A slot to add a checkout step.
- A slot to add custom fields or code to checkout steps.
- A slot to add Cart total lines.

With that, we should also start considering what kind of components we are going to expose and make public, this is either as a dependency or external.

Some things that doesn't work as a slot, but as a filter:
- Adding extra regular inputs to existing forms, this should be a filter.
- Adding custom validation for certain things, this should be an event subscription or a filter that is associated with custom inputs.

Something that should be handled in the backend:
- Adding filters and actions for extending API response (when it is stable), for reference I'm not suggesting we open the Store API for public use, but for extending current functionality, which I think is different.

With that, to start this, we should probably move some of our code to be using those extensions and slots, and start documenting how others can do this.

**Comment by nerrad** [comment]
Thanks for starting this topic Nadir! I agree slot/fill can be a useful pattern for adding extensibility to the blocks (and is likely something we'll implement). However before we even get to the _how_ of doing this, we should first work out the _what_ (and why) we want for extensibility. To that end, I've already created #1216 and #1217 to contain that discussion.

I removed the documentation label only because I don't think we really have anything to document yet. Let's keep this issue for discussing potentially using slot/fill as a pattern for extensibility once we're finished #1216 and #1217.
